### PR TITLE
Add d2l-navigation-separator reference and remove unused CSS

### PR DESCRIPTION
**Pull request by jeroach**
`d2l-navigation` PR: https://github.com/BrightspaceUI/navigation/pull/3
`platformTools navbars` PR: https://github.com/BrightspaceUI/navigation/pull/3

**Navigation PR needs to be merged before this passes CI.**

### CODE REVIEW

File: sass/navigation/divider.scss [hunk 1]
```diff
@@ -1,17 +1,7 @@
-@import '../../bower_components/d2l-colors/d2l-colors.scss';
 @import 'common.scss';
 
 .d2l-navigation-s-divider {
```
Comment (dlockhart) [inline sass/navigation/divider.scss]: I think we might be able to totally get rid of this class name (and the `<div>`) and replace it with something like:

```
.d2l-navigation-s d2l-navigation-separator {
    @media (max-width: $d2l-navigation-bp-separators) {
        display: none;
    }
}
```

Comment (jeroach) [reply sass/navigation/divider.scss]: Yep, this works! Thanks!

File: sass/navigation/divider.scss [hunk 2]
```diff
@@ -1,17 +1,9 @@
...
-		color: $d2l-color-mica;
-	}
-	@media (max-width: $d2l-navigation-bp-separators) {
- 		display: none;
- 	}
-}
-.d2l-navigation-s-divider-no-margin {
-	margin: 0;
+
+    @media (max-width: $d2l-navigation-bp-separators) {
```
Comment (svanherk) [inline sass/navigation/divider.scss]: Super minor: The formatting here went a little weird

Comment (jeroach) [reply sass/navigation/divider.scss]: Hmmm, Atom doesn't seem to show a difference between the formatting of this file and other `.scss` files. Perhaps it's just the GitHub diff?

**Pull request merged**